apps/src/code-studio/pd/foorm/editor/form/FormEditorManager.jsx

```diff
@@ -106,31 +104,33 @@ class FormEditorManager extends React.Component {
           hasLoadError: true
         });
       });
-  }
+  };
 
-  initializeEmptyCodeMirror = () => {
-    this.props.setLastSaved(null);
-    this.props.setSaveError(null);
-    this.updateFormData({
+  // For loading buttons
+  resetSelectedData() {
+    this.props.setFormData({
       questions: {},
       published: null,
       formName: null,
       formVersion: null,
       formId: null
     });
-    this.setState({
-      showCodeMirror: true,
-      hasLoadError: false
-    });
-  };
+  }
+
+  // For loading buttons
+  showCodeMirror() {
+    this.setState({showCodeMirror: true});
+  }
 
+  // For loading buttons
   updateFormData(formData) {
     this.props.setFormData(formData);
     this.props.setHasJSONError(false);
     this.props.setLastSavedFormQuestions(formData['questions']);
     this.props.resetCodeMirror(formData['questions']);
   }
 
+  // for preview
   // use debounce to only call once per second
   fillFormWithLibraryItems = _.debounce(
```

Comment:
this should be a function instead of a class property

Reply:
Let's look at this tomorrow, I cannot for the life of me figure out how to bind this appropriately!

Reply:
should just be
```
constructor(props) {
...
  this.fillFormWithLibraryItems = this.fillFormWithLibraryItems.bind(this);
...
}

fillFormWithLibraryItems() {
  _.debounce(...)
}
```
does that not work?

Reply:
For anyone following along at home, check out the final merged code to see where we landed on this :).